geoapi/routes/streetview_mapillary_auth.py

```diff
@@ -0,0 +1,180 @@
+import requests
+import secrets
+from datetime import datetime, timedelta
+from urllib.parse import urlencode
+
+from geoapi.services.users import UserService
+from geoapi.settings import settings
+from geoapi.services.streetview import StreetviewService
+from geoapi.models import Streetview
+from geoapi.log import logging
+from geoapi.utils.decorators import jwt_decoder
+from geoapi.db import db_session
+from geoapi.exceptions import AuthenticationIssue
+from geoapi.utils.client_backend import (
+    validate_referrer_url,
+    get_client_url,
+    get_deployed_geoapi_url,
+)
+from flask_restx import Namespace, Resource
+from flask import request, redirect, session, jsonify, abort
+import jwt
+
+
+def get_auth_state():
+    return secrets.token_hex(24)
+
+
+logger = logging.getLogger(__name__)
+
+api = Namespace("streetview_auth", path="/streetview/auth")
+
+
+# TODO: this is a work around for lack of sessions
```

Comment:
@sophia-massie @rstijerina

Since we're using the Tapis JWT instead of sessions, I needed a way to redirect users to `/mapillary/login` without exposing the JWT in logs or browser history.

- Sessions: not implemented  
- JWT as query param: insecure (logs, history, etc.)

So `/mapillary/prepare` creates a short-lived token, and `/mapillary/login` uses it to start the Mapillary OAuth flow.